app/web/src/newhotness/explore_grid/ExploreGridRow.vue

```diff
@@ -276,10 +316,56 @@ const props = defineProps<{
 const exploreContext = inject<ExploreContext>("EXPLORE_CONTEXT");
 assertIsDefined<ExploreContext>(exploreContext);
 
-const renderLastPath = (path: string) => {
-  const last = path.split("/").at(-1) ?? "unknown";
-  return last.charAt(0).toLocaleUpperCase() + last.slice(1);
-};
+const ctx = useContext();
+
+const key = useMakeKey();
+const args = useMakeArgs();
+
+const enableDefaultSubSourceValuesQuery = computed(
+  () => ctx.queriesEnabled && props.row.type === "defaultSubHeader",
+);
+const defaultSubSourceValuesQuery = useQuery<
+  { prop: string; value: string } | undefined
+>({
+  enabled: enableDefaultSubSourceValuesQuery,
+  queryKey: key(
+    EntityKind.AttributeTree,
+    props.row.type === "defaultSubHeader" ? props.row.componentId : "",
+    "DefaultSubHeader",
+  ),
+  queryFn: async () => {
+    if (props.row.type !== "defaultSubHeader") {
```

Comment:
it feels to me like this should also be whether or not the query is enabled? it looks like a "we dont have data so just get out"?

Reply:
The query will only be enabled if prop.rows.type is defaultSubHeader, but we also need to type assert here to get access to the row fields. Hence the return.

Reply:
ohhhh its for the typing OK 🫡 maybe just a comment then `// this is not reachable`